encode_test.go

```diff
@@ -656,6 +656,17 @@ func (s *S) TestSetIndent(c *C) {
 	c.Assert(buf.String(), Equals, "a:\n        b:\n                c: d\n")
 }
 
+func (s *S) TestSequenceUndent(c *C) {
+	var buf bytes.Buffer
+	enc := yaml.NewEncoder(&buf)
+	enc.SetSequenceUndent(true)
+	err := enc.Encode(map[string]interface{}{"a": []interface{}{"b", "c", "d"}})
+	c.Assert(err, Equals, nil)
+	err = enc.Close()
```

Comment:
Can we use defer? Also, do we need a test for the `close` func?

Reply:
I don't think I can use defer here as `enc.Close()` flushes the buffer and I can't assert the change until after that is done. Defer would mean it runs when the surrending function returns.


As for a seperate test for that, I'm not sure we gain additional confidence as it is implicitly tested in both my new test and all the old encoder settings tests.